config/initializers/ldap_authenticatable.rb

```diff
@@ -3,15 +3,24 @@ module Strategies
     class LdapAuthenticatable < Authenticatable
 
       def authenticate!
-        if valid_credentials?("#{username}@chapman.edu", password)
+
+        if valid_credentials?("#{username}@chapman.edu", password)          
           success! User.create_or_update_from_active_directory(username)
         else
           fail(:invalid_login)
         end
-      rescue UnexpectedActiveDirectoryFormat, ChapmanIdentityNotFound
+      rescue UnexpectedActiveDirectoryFormat
+        fail(:invalid_login)
+      #if identity issue occurs in create_or_update_from_active_directory method ChapmanandIdentityNotFound error is raised and with user information reported to bugsnag in rescue
+      rescue ChapmanIdentityNotFound
+        user = User.where(email: "#{username}@chapman.edu")
+        identity_info = User.lookup_in_active_directory(username)
+
+        notify_bugsnag(user: user, response: identity_info)
         fail(:invalid_login)
```

Comment:
I don't like that we're using the same failure message for invalid credentials and invalid identity info. It's misleading to users if they just think they're entering the wrong username and password. If you look at Inside, the error messages are different. Please add an error message like the `invalid_identity_info` message in Inside to use inside this `rescue` (and add it to `config/locales/devise.en.yml`) instead of the `invalid_login` message.

Reply:
Added `invalid_identity_info` to `config/locales/devise.en.yml` with same message "Invalid identity info" as inside `config/locales/devise.en.yml`